pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,188 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This submodule contains the ParametrizedHamiltonian class
+"""
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.ops.qubit.hamiltonian import Hamiltonian
+from pennylane.wires import Wires
+
+
+# pylint: disable= too-many-instance-attributes
+class ParametrizedHamiltonian:
+    r"""Callable object holding the information representing a parametrized Hamiltonian.
+
+    Passing parameters to the ``ParametrizedHamiltonian`` returns an :class:`~pennylane.operation.Operator` representing the Hamiltonian
+    for that set of parameters.
+
+    The Hamiltonian can be represented as a linear combination of other operators, e.g.,
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j`, where the :math:`v` are trainable parameters,
+    and t is time.
+
+    For example, a time-dependent ``ParametrizedHamiltonian`` with a single trainable parameter, :math:`a`, could be :math:`H = 2 X_1 X_2 + \sin(a t) Y_1 Y_2`
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be constants or
+            parametrized functions. All functions passed as ``coeffs`` must have the signature ``(params, t)``
+            with ``params`` being trainable parameters and ``t`` being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same length as ``coeffs``
+
+    A ``ParametrizedHamiltonian`` is callable with the fixed signature ``H(params, t)`` that returns an
+    ``Operator`` representing an instance of the Hamiltonian with the specified parameter values.
+
+    **Example:**
+
+    .. code-block:: python3
+
+        coeffs = [2, lambda v, t: v[0] * jnp.sin(v[1] * t)]
+        observables =  [qml.PauliX(0), qml.PauliY(1)]
+        H = ParametrizedHamiltonian(coeffs, observables)
+
+    >>> H(jnp.ones(2), 1.)
+    (2*(PauliX(wires=[0]))) + (0.8414710164070129*(PauliY(wires=[1])))
+
+    A ``ParametrizedHamiltonian`` can be created by passing a list of coefficients (scalars or functions), as well as
+    a list of corresponding observables. The functions must all have signatures, ``(params, t)`` with ``params`` being
+    trainable parameters and ``t`` being time.
+
+    >>> def f1(params, t): return np.sin(params[0]*t)
+    >>> def f2(params, t): return params[1] * np.cos(t)
+
+    The functions, along with scalar coefficients, can then be used to initialize a ``ParametrizedHamiltonian``,
+    which will be split into a fixed and parametrized term. The fixed term is an ``Operator``, while the parametrized
+    term must be initialized with concrete parameters to obtain an ``Operator``.
+
+    .. code-block:: python3
+
+        coeffs = [2, f1, f2]
+        obs = [qml.PauliX(0)@qml.PauliX(1), qml.PauliY(0)@qml.PauliY(1), qml.PauliZ(0)@qml.PauliZ(1)]
```

Comment:
```suggestion
        obs = [qml.PauliX(0) @ qml.PauliX(1), qml.PauliY(0) @ qml.PauliY(1), qml.PauliZ(0) @ qml.PauliZ(1)]
```
Without the spaces the @qml renders differently
![image](https://user-images.githubusercontent.com/43949391/212965142-610e7e39-3801-45fc-8010-102e837afe69.png)